Analysis/Tasks/PWGHF/HFCandidateCreator3Prong.cxx

```diff
@@ -170,18 +182,38 @@ struct HFCandidateCreator3ProngMC {
       // Λc± → p± K∓ π±
       if (result == N3ProngDecays) {
         //Printf("Checking Λc± → p± K∓ π±");
-        if (RecoDecay::getMatchedMCRec(particlesMC, std::move(arrayDaughters), 4122, array{+kProton, -kKPlus, +kPiPlus}, true, &sign) > -1) {
+        auto indexRecLc = RecoDecay::getMatchedMCRec(particlesMC, std::move(arrayDaughters), 4122, array{+kProton, -kKPlus, +kPiPlus}, true, &sign, -1);
```

Comment:
Is it needed to keep the tree depth unrestricted?

Reply:
It can be fixed to 2. I already checked that the result is the same.